csharp-basic-topics/ConvertingReadOnlyMemoryToByteArrayInCSharp/ConvertingReadOnlyMemoryToByteArrayInCSharp/Program.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var numsResult = ReadOnlyMemoryOfIntToByteArray();
+        Console.WriteLine(Convert.ToBase64String(numsResult)); // output: AQAAAAIAAAADAAAA
+
+        var charsResult = ReadOnlyMemoryOfCharToByteArray();
+        Console.WriteLine(Convert.ToBase64String(charsResult)); // output: ZgBvAG8A
+
+        string documentsPath = Environment.GetFolderPath(Environment.SpecialFolder.Personal);
+        var filePath = Path.Combine(documentsPath, "hello.txt");
+        var content = "Hello, World!";
+        SaveFile(filePath, content); // saves hello.txt in the Documents folder
+
+        var hash = HashPassword("foo");
+        Console.WriteLine(Convert.ToBase64String(hash)); // output: LCa0a2j/xo/5m0U8HTBBNBNCLXBkg7+g+YpeiGJm564=
+    }
+
+    public static byte[] ReadOnlyMemoryOfIntToByteArray()
+    {
+        ReadOnlyMemory<int> numbers = new int[] { 1, 2, 3 };
+        var numsByteArray = MemoryMarshal.AsBytes(numbers.Span).ToArray();
+        return numsByteArray;
+    }
+
+    public static byte[] ReadOnlyMemoryOfCharToByteArray()
+    {
+        ReadOnlyMemory<char> characters = "foo".AsMemory();
+        var charsByteArray = MemoryMarshal.AsBytes(characters.Span).ToArray();
+        return charsByteArray;
+    }
+
+    public static void SaveFile(string path, string content)
+    {
+        ReadOnlyMemory<byte> readOnlyMemory = Encoding.UTF8.GetBytes(content);
+        using (var stream = new FileStream(path, FileMode.Create, FileAccess.Write))
+        {
+            byte[] byteArray = readOnlyMemory.ToArray();
+            stream.Write(byteArray, 0, byteArray.Length);
+        }
+    }
+
+    public static byte[] HashPassword(string password)
+    {
+        ReadOnlyMemory<byte> passwordMemory = Encoding.UTF8.GetBytes(password);
+        using (var hashAlgorithm = SHA256.Create())
+        {
+            byte[] byteArray = passwordMemory.ToArray();
+            return hashAlgorithm.ComputeHash(byteArray);
+        }
+    }
```

Comment:
The direction of these examples is good, but some users may comment that since Encoding.UTF8.GetBytes() is already returning a byte[], we are in essence just copying one array to another array in order to pass the copied array into the methods.
So the idea and technique/thought here is good. Maybe for the file writing example we could make this a generic method that takes a ReadOnlyMemory<T> (which also means that we can make this method async since we can't use ReadOnlySpan as an argument to an async). We could then use the Marshal class to marshall the memory to bytes (as we did previously) and then call the WriteAsync method to write the bytes. We can use the unmanaged constraint to ensure that T is bittable.